packages/syft/src/syft/core/node/new/action_object.py

```diff
@@ -321,57 +370,247 @@ def syft_is_property(self, obj: Any, method: str) -> bool:
+            if isinstance(uid, (LineageID, UID)):
+                kwarg_ids[k] = LineageID(uid)
+                continue
+            if isinstance(uid, (ActionObjectPointer, ActionObject)):
+                kwarg_ids[k] = uid
+                continue
+            raise ValueError(
+                f"Invalid kwargs type {type(uid)}. Must be [UID, LineageID, ActionObject or ActionObjectPointer]"
+            )
+
+        action = Action(
+            path=path,
+            op=op,
+            remote_self=LineageID(remote_self),
+            args=arg_ids,
+            kwargs=kwarg_ids,
+        )
+        return action
+
+    def syft_make_method_action(
+        self,
+        op: str,
+        args: Optional[List[Union[UID, ActionObjectPointer]]] = None,
+        kwargs: Optional[Dict[str, Union[UID, ActionObjectPointer]]] = None,
+    ) -> Action:
+        """Generate new method action from the current object.
+
+        Parameters:
+            op: str
+                The method to be executed from the remote object.
+            args: List[LineageID]
+                `op` args
+            kwargs: Dict[str, LineageID]
+                `op` kwargs
+        Returns:
+            Action object
+
+        Raises:
+            ValueError: For invalid args or kwargs
+            PydanticValidationError: For args and kwargs
+        """
+        path = self.syft_get_path()
+        return self.syft_make_action(
+            path=path, op=op, remote_self=self.syft_lineage_id, args=args, kwargs=kwargs
+        )
+
+    def syft_get_path(self) -> str:
+        """Get the type of the underlying object"""
+        if isinstance(self, AnyActionObject) and self.syft_internal_type:
+            return f"{type(self.syft_action_data).__name__}"  # avoids AnyActionObject errors
+        return f"{type(self).__name__}"
+
+    def syft_remote_method(
+        self,
+        op: str,
+    ) -> Callable:
+        """Generate a Callable object for remote calls.
+
+        Parameters:
+            op: str
+                he method to be executed from the remote object.
+
+        Returns:
+            A function
+        """
+
+        def wrapper(
+            *args: Optional[List[Union[UID, ActionObjectPointer]]],
+            **kwargs: Optional[Dict[str, Union[UID, ActionObjectPointer]]],
+        ) -> Action:
+            return self.syft_make_method_action(op=op, args=args, kwargs=kwargs)
+
+        return wrapper
+
     def send(self, client: SyftClient) -> Self:
+        """Send the object to a Syft Client"""
+
         return client.api.services.action.set(self)
 
     def get_from(self, client: SyftClient) -> Any:
+        """Get the object from a Syft Client"""
+
         return client.api.services.action.get(self.id).syft_action_data
 
     @staticmethod
     def from_obj(
         syft_action_data: Any,
         id: Optional[UID] = None,
         syft_lineage_id: Optional[LineageID] = None,
-    ) -> ActionObject:
```

Comment:
personally I am not sold on the `ActionObject.from_obj(x).ok()` dev experience. Curious to hear what others thinks

Reply:
not sure if users would use it directly though, or always work with something that is generating those

Reply:
Sure, but at some point, the library should stick to some error handling strategy. And Result seems to be already used a lot in the library.